transcom/mymove
# B 20495 move search and customer search trailing space int

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20495)

## Summary
The issue was that when an office user with the role of TOO, QAE or service counselor would copy and paste a Move code or DODID it would sometimes have one or more trailing spaces. When submitting for the search the code or DODID would be submitted to the API call it would have these trailing white spaces and would cause the api call to fail. The solution was to trim that white space so that only the actual move code and actual dodid without any trailing spaces would be searched.

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the office
2. Login as a TOO, TIO or QAE
3. Search for a move using the move code and add at least one trailing white space (This should now not crash the application)
4. Search for a customer using the DODID and add at least one trailing white space (This should now not crash the application)

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
Notice the trailing spaces. Before this would crash.
![image](https://github.com/user-attachments/assets/117b9d55-3a37-478a-82c7-e2c67d2ebb66)
